cmd/builder/internal/builder/templates/main.go.tmpl

```diff
@@ -16,8 +21,25 @@ func main() {
 		Description: "{{ .Distribution.Description }}",
 		Version:     "{{ .Distribution.Version }}",
 	}
+	providers := []confmap.Provider{
+		{{- range .Providers}}
+		{{.Name}}.NewWithSettings(confmap.ProviderSettings{}),
+		{{- end}}
+	}
+	set := otelcol.CollectorSettings{
+		BuildInfo: info,
+		Factories: components,
+		ConfigProviderSettings: otelcol.ConfigProviderSettings{
+			ResolverSettings: confmap.ResolverSettings{
+				Providers:  makeMapProvidersMap(providers...),
+				Converters: []confmap.Converter{
+					expandconverter.New(confmap.ConverterSettings{}),
+				},
+			},
+		},
+	}
```

Comment:
Ditto, this API will only work with new enough versions of the builder, so I would guard this with a check to `otelcol_version`